File: forge/templates/apps/flutter-frontend-template/{{project_slug}}/lib/src/features/chat/chat_constants.dart
class ChatModel {
  const ChatModel({required this.id, required this.label});
  final String id;
  final String label;
}

const availableModels = <ChatModel>[
  ChatModel(id: 'gpt-4.1', label: 'GPT-4.1'),
  ChatModel(id: 'gpt-4.1-mini', label: 'GPT-4.1 Mini'),
  ChatModel(id: 'claude-sonnet-4', label: 'Claude Sonnet 4'),
];

enum ApprovalMode { defaultMode, bypass }

extension ApprovalModeX on ApprovalMode {
  String get label =>
      this == ApprovalMode.bypass ? 'Bypass' : 'Default';

  String get wireValue =>
      this == ApprovalMode.bypass ? 'bypass' : 'default';

  static ApprovalMode fromWire(String? value) =>
      value == 'bypass' ? ApprovalMode.bypass : ApprovalMode.defaultMode;
}

const defaultModelId = 'claude-sonnet-4';
const defaultApprovalMode = ApprovalMode.defaultMode;

File: forge/templates/apps/svelte-frontend-template/template/src/lib/features/chat/chat.constants.ts
/** Available LLM models the user can pick from in the chat panel header. */
export const AVAILABLE_MODELS = [
	{ id: 'gpt-4.1', label: 'GPT-4.1' },
	{ id: 'gpt-4.1-mini', label: 'GPT-4.1 Mini' },
	{ id: 'claude-sonnet-4', label: 'Claude Sonnet 4' }
] as const;

export type ModelId = (typeof AVAILABLE_MODELS)[number]['id'];

/** Approval modes — `default` = ask before running tools, `bypass` = auto-approve. */
export const APPROVAL_MODES = [
	{ id: 'default', label: 'Default' },
	{ id: 'bypass', label: 'Bypass' }
] as const;

export type ApprovalMode = (typeof APPROVAL_MODES)[number]['id'];

export const DEFAULT_MODEL: ModelId = 'claude-sonnet-4';
export const DEFAULT_APPROVAL: ApprovalMode = 'default';
